scripts/download_incr_DB.py

```diff
@@ -198,18 +236,39 @@ def calculate_multipart_etag(source_path, chunk_size):
 def run():
 	while (True):
 		try:
+			currTxBlk = -1
 			if(UploadLock() == False):
+				currTxBlk = GetCurrentTxBlkNum()
+				if(currTxBlk < 0): # wait until current txblk is known
+					time.sleep(1)
+					continue
 				print("[" + str(datetime.datetime.now()) + "] Started downloading entire persistence")
 				GetEntirePersistenceFromS3()
 			else:
-				continue
-
-			if(UploadLock() == True):
-				print("Upload has been triggered. Downloading StateDelta now can cause inconsistent data. will restart again..")
+				time.sleep(1)
 				continue
 
 			print("Started downloading State-Delta")
 			GetStateDeltaFromS3()
+			#time.sleep(30) // uncomment it for test purpose.
+			newTxBlk = GetCurrentTxBlkNum()
+			if(currTxBlk < newTxBlk):
+				# To get new files from S3 if new files where uploaded in meantime
+				while(UploadLock() == True):
+					time.sleep(1)
+			else:
+				break
+			if(currTxBlk < newTxBlk):
```

Comment:
I think it's not necessary to re-check again here?  Line 254 check should still apply right?